torchmetrics/detection/map.py

```diff
@@ -565,6 +565,12 @@ def _calculate(self, class_ids: List) -> Tuple[Dict, MAPMetricResults, MARMetric
         recall = -torch.ones((nb_iou_thrs, nb_classes, nb_bbox_areas, nb_max_det_thrs))
         scores = -torch.ones((nb_iou_thrs, nb_rec_thrs, nb_classes, nb_bbox_areas, nb_max_det_thrs))
 
+        # move tensors if necessary
+        if self.max_detection_thresholds.device != self.device:
+            self.max_detection_thresholds = self.max_detection_thresholds.to(self.device)
```

Comment:
this doesn't have to be a if-statement. This check is already done inside .to(), so it will be a no-op if the tensor is already on that device.

Reply:
Thanks, fixed it.